### PR TITLE
loWeb Refactor

### PR DESCRIPTION
**Pull request by Dumeng**
Refactor the loWeb module to a MediaService object.

Next step, after necessary testing and bug fixes, we can move the logic to the background thread.
Also, we can refactor all providers from JS objects to native JS classes. This will make new providers easier to be added.

### CODE REVIEW

**Comment by listen1** [comment]
bug report: toggle nowplaying not work.

after open now playing and click music cover to toggle, page has no response.

**Review by listen1 — CHANGES_REQUESTED** [review]
details pleae refer to bug report in PR

**Comment by listen1** [comment]
sorry for review delay, I'll process it ASAP.

File: js/loweb.js [hunk 1]
```diff
@@ -1,378 +1,302 @@
...
-        const srcData = localStorage.getObject(source).tracks;
-        tarData.forEach((tarTrack) => {
-          if (!srcData.find((srcTrack) => srcTrack.id === tarTrack.id)) {
-            myplaylist.add_myplaylist(source, tarTrack);
-          }
-        });
-        return {
-          success: (fn) => fn(),
-        };
+  bootstrapTrack(sound, track, playerSuccessCallback, playerFailCallback) {
```
Comment (listen1) [inline js/loweb.js:247]: auto choose source seems to have some problem. see bug report for more details

Comment (Dumeng) [reply js/loweb.js:247]: The tricky usage of Promise makes the `playerFailCallback` is always called.

Comment (listen1) [reply js/loweb.js:247]: nice, problem solved.

**Pull request merged**